### PR TITLE
CR-1090469 and CR-1090166

### PR DESCRIPTION
**Pull request by AShivangi**
- CR-1090469 ASTeR TC 21.03 - u30 - Improve error message when p2p is not supported, but is attempted to be enabled
- CR-1090166 xbmgmt2 not seeing the U30 SC
```
$ xbmgmt -new examine
-----------------------------------------------------
 Invoking next generation of the xbmgmt application
-----------------------------------------------------
-----------------------------------------------
1/16 [0000:6d:00.0] : xilinx_u30_gen3x4_base_1
-----------------------------------------------
Flash properties
  Type                 : qspi_ps_x2_single
  Serial Number        : XFL1BRW4DUYR

Flashable partitions running on FPGA
  Platform             : xilinx_u30_gen3x4_base_1
  SC Version           : 6.3.5
  Platform ID          : BDB6119B-AC58-C962-1062-9DE857A4CE00
  Interface UUID       : 265FF36F-14F4-A973-3531-A7BF8EE32083

Flashable partitions installed in system
  Platform             : xilinx_u30_gen3x4_base_1
  SC Version           : 6.3.5
  Platform ID          : 0xbdb6119bac58c962

----------------------------------------------------
-----------------------------------------------
2/16 [0000:6c:00.0] : xilinx_u30_gen3x4_base_1
-----------------------------------------------
Flash properties
  Type                 : qspi_ps_x2_single
  Serial Number        : XFL1BRW4DUYR

Flashable partitions running on FPGA
  Platform             : xilinx_u30_gen3x4_base_1
  SC Version           : 6.3.5(FIXED)
  Platform ID          : BDB6119B-AC58-C962-1062-9DE857A4CE00
  Interface UUID       : 265FF36F-14F4-A973-3531-A7BF8EE32083

Flashable partitions installed in system
  Platform             : xilinx_u30_gen3x4_base_1
  SC Version           : 6.3.5
  Platform ID          : 0xbdb6119bac58c962

```

### CODE REVIEW

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp [hunk 1]
```diff
@@ -537,7 +537,7 @@ bool XMC_Flasher::isXMCReady()
 
 bool XMC_Flasher::isBMCReady()
 {
-    bool bmcReady = (BMC_MODE() == 0x1);
+    bool bmcReady = (BMC_MODE() == 0x1) || (BMC_MODE() == 0x5);
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:540]: Could you please add a comment indicating what the values 0x1 and 0x5 represent.

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp [hunk 2]
```diff
@@ -556,3 +556,16 @@ bool XMC_Flasher::hasSC()
     } catch (...) {}
     return sc_presence;
 }
+
+bool XMC_Flasher::fixedSC()
+{
+    bool is_sc_fixed = false;
+    std::string errmsg;
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:563]: Is this string needed?

**Review by rozumx — APPROVED** [review]
Approved with minor changes requested.

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:540]: This code will change with the upcoming cleanup. This code should use enum but I wanted to keep the code change to the minimum for backporting

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:563]: No it's not. Good catch. I'll fix it with the upcoming cleanup :)

**Pull request merged**